Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -827,12 +827,26 @@ void ContainerNode::removeChildren()
 
     Ref protectedThis { *this };
     NodeVector removedChildren;
-    removeAllChildrenWithScriptAssertion(ChildChange::Source::API, removedChildren);
-
+#if PLATFORM(MAC)
+    if (this != &rootNode() && this->document().settings().opportunisticallyDeleteNodes() && this->document().settings().opportunisticSweepingAndGarbageCollectionEnabled()) {
+        removeAllChildrenWithScriptAssertion(ChildChange::Source::API, removedChildren, DeferChildrenChanged::No, CanDelayNodeDeletion::Yes);
+        delayDeletingRemovedChildren(WTFMove(removedChildren));
+    } else {
+        // do not unintentionally keep a document alive.
```

Comment:
What is this comment about? removeAllChildrenWithScriptAssertion is nothing to do with Document.